src/gazebo_mavlink_interface.cpp

```diff
@@ -516,19 +516,28 @@ void GazeboMavlinkInterface::ImuCallback(ImuPtr& imu_message) {
     sensor_msg.abs_pressure = pressure_msl / pressure_ratio;
 
     // generate Gaussian noise sequence using polar form of Box-Muller transformation
-    // http://www.design.caltech.edu/erik/Misc/Gaussian.html
-    double x1, x2, w, y1, y2;
-    do {
-     x1 = 2.0 * (rand() * (1.0 / (double)RAND_MAX)) - 1.0;
-     x2 = 2.0 * (rand() * (1.0 / (double)RAND_MAX)) - 1.0;
-     w = x1 * x1 + x2 * x2;
-    } while ( w >= 1.0 );
-    w = sqrt( (-2.0 * log( w ) ) / w );
-    y1 = x1 * w;
-    y2 = x2 * w;
+    double x1, x2, w, y1;
+    static double y2;
```

Comment:
Can you avoid `static` here and make these a class attribute?

Reply:
Fixed